app/src/pages/clinic.tsx

```diff
@@ -1,27 +1,51 @@
+import { useAppContext } from '@context/state'
 import clinics from '@public/clinic-output/clinics-with-ids.json'
 import type { GetServerSideProps, NextPage } from 'next'
 import { useTranslation } from 'next-i18next'
 import { serverSideTranslations } from 'next-i18next/serverSideTranslations'
 import Image from 'next/image'
 import Link from 'next/link'
-import { ChangeEvent, FormEvent, useState } from 'react'
+import { ChangeEvent, FormEvent, useEffect, useState } from 'react'
 
 import Alert from '@components/Alert'
 import ButtonLink from '@components/ButtonLink'
 
-const Clinic: NextPage = () => {
+interface Props {
+  previousRoute: string
+}
+
+const Clinic: NextPage<Props> = (props: Props) => {
   const { t } = useTranslation('common')
+  const { session, setSession } = useAppContext()
   const [expandList, setExpandList] = useState<boolean>(false)
   const numberOfClinicsToReturn = 8
-  const [filteredClinics, setFilteredClinics] = useState<
-    (typeof clinics[0] | undefined)[]
-  >([])
   const [selectedClinic, setSelectedClinic] = useState<
     typeof clinics[0] | undefined
-  >(undefined)
+  >(session?.clinic)
+  const [filteredClinics, setFilteredClinics] = useState<
+    (typeof clinics[0] | undefined)[]
+  >(session?.clinic ? [selectedClinic] : [])
   const [search, setSearch] = useState('')
   const [searchError, setSearchError] = useState<boolean>(false)
   const [zipValidationError, setZipValidationError] = useState<boolean>(false)
+  const [continueBtn, setContinueBtn] = useState<{
+    label: string
+    route: string
+    width: string
+  }>({ label: t('Clinic.button'), route: '/contact', width: '251px' })
+
+  useEffect(() => {
```

Comment:
Why do we need to run this in a `useEffect()`? I don't think the dependencies change except for when the page loads, right?

Reply:
@karinamzalez Please add a comment to this line to say something like "We are using `useEffect()` because we want to make sure the props provided by `getServerSideProps()` are reliably loaded into the page."

Reply:
Please also add a "TODO research" comment

Reply:
See https://github.com/navapbc/wic-mt-demo-project-eligibility-screener/blob/78f2fb184616f5a56b22b0b62b9447703691ad49/app/src/pages/eligibility.tsx#L38 for a case where we almost certainly _do_ want to use `useEffect()` because it is dependent on user interaction on that page.